src/Standards/Squiz/Sniffs/Arrays/ArrayDeclarationSniff.php

```diff
@@ -620,20 +619,32 @@ public function processMultiLineArray($phpcsFile, $stackPtr, $arrayStart, $array
                     continue;
                 }
 
-                if ($lastValueLine !== false && $tokens[$value['value']]['line'] === $lastValueLine) {
+                $valuePointer = $value['value'];
+
+                if ($tokens[$valuePointer]['line'] === $tokens[$stackPtr]['line']) {
+                    $error = 'The first value in a multi-value array must be on a new line';
+                    $fix   = $phpcsFile->addFixableError($error, $stackPtr, 'FirstValueNoNewline');
+                    if ($fix === true) {
+                        $phpcsFile->fixer->addNewlineBefore($valuePointer);
+                    }
+                }
+
+                $previous = $phpcsFile->findPrevious([T_WHITESPACE, T_COMMA], ($valuePointer - 1), ($arrayStart + 1), true);
+                $previousIsWhitespace = $tokens[($valuePointer - 1)]['code'] === T_WHITESPACE;
+                if ($previous !== false && $tokens[$previous]['line'] === $tokens[$valuePointer]['line']) {
                     $error = 'Each value in a multi-line array must be on a new line';
-                    $fix   = $phpcsFile->addFixableError($error, $value['value'], 'ValueNoNewline');
+                    $fix   = $phpcsFile->addFixableError($error, $valuePointer, 'ValueNoNewline');
                     if ($fix === true) {
-                        if ($tokens[($value['value'] - 1)]['code'] === T_WHITESPACE) {
-                            $phpcsFile->fixer->replaceToken(($value['value'] - 1), '');
+                        if ($previousIsWhitespace === true) {
```

Comment:
I think the preferred syntax is
```
$prevToken = $phpcsFile->findPrevious(T_WHITESPACE, ($token - 1), null, true);

for ($i = ($prevToken + 1); $i < $token; $i++) {
    $phpcsFile->fixer->replaceToken($i, '');
}

$phpcsFile->fixer->addNewLineBefore($token);
```

Reply:
Again, please explain how would that work.

Anyway, the variable is needed, as it is used in two places, not just here.